modules/lambda-registrator/variables.tf

```diff
@@ -11,17 +11,15 @@ variable "consul_http_addr" {
 variable "consul_ca_cert_path" {
   description = "The Parameter Store path containing the Consul server CA certificate."
   type        = string
-  default     = ""
```

Comment:
I believe this is the CA Cert for the Consul server's HTTPS, and that CA cert is actually not required for HCP Consul servers, which have HTTPS urls with proper signed cert.

Reply:
I could be thinking about this wrong, so feel free to pass down some wisdom: This value is generated from the certificate I present for storage in parameter store, which is the ca cert I retrieve from Consul. In my/HCP's case, it's the [`hcp_consul_cluster.consul_ca_file`](https://registry.terraform.io/providers/hashicorp/hcp/latest/docs/resources/consul_cluster#consul_ca_file) output of the chainfile. So what I'm passing to parameter store is the capability for the registrator/another consul service to identify itself to HCP, instead of presenting a newly-generated certificate (Which is correct by my understanding to your point, we don't need to generate a brand new cert; HCP does this automagically). This, along with the bootstrap token allows for a successful join to the cluster, otherwise HTTPS verification would fail, I think?

Reply:
What Paul is saying is that you only need a custom CA cert if the Consul API is signed by a custom CA. Just like if you want to `curl https://google.com` you can do this and it will work, you don't need to use `curl --cacert=blah https://google.com`. (What happens under the hood is that the container ships with a bunch of publicly trusted root certs).

So in the case of HCP, the API uses a publicly trusted root cert just like google.com, so you don't have to pass in the cert you get out of the HCP terraform output.

Reply:
Appreciate the clarity @lkysow 😸